config/crd/multicluster.x-k8s.io_serviceimports.yaml

```diff
@@ -1,183 +1,351 @@
+              Supports "ClientIP" and "None". Defaults to "None".
+              Reflects the `.spec.sessionAffinity` configuration of the underlying
+              exported Service. Ignored when exported Service type is Headless. If
+              exported Services of the same name and namespace in a given ClusterSet have
+              differing session affinity configuration, a "Conflict" status condition
+              SHOULD be reported in ServiceExport status.
+              More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+            type: string
+          sessionAffinityConfig:
+            description: |-
+              sessionAffinityConfig contains session affinity configuration.
+              Reflects the `.spec.sessionAffinityConfig` configuration of the underlying
+              exported Service. Ignored when exported Service type is "Headless". If
+              exported Services of the same name and namespace in a given ClusterSet have
+              differing session affinity configuration, a "Conflict" status condition
+              SHOULD be reported in ServiceExport status.
+            properties:
+              clientIP:
+                description: clientIP contains the configurations of Client IP based
+                  session affinity.
+                properties:
+                  timeoutSeconds:
+                    description: |-
+                      timeoutSeconds specifies the seconds of ClientIP type session sticky time.
+                      The value must be >0 && <=86400(for 1 day) if ServiceAffinity == "ClientIP".
+                      Default value is 10800(for 3 hours).
+                    format: int32
+                    type: integer
+                type: object
+            type: object
+          type:
+            description: |-
+              type defines the type of this service.
+              Must be "ClusterSetIP" or "Headless".
+              The "ClusterSetIP" type reflects exported Service(s) with type ClusterIP
+              and the "Headless" type reflects exported Service(s) with type Headless.
+              A ServiceImport with type ClusterSetIP SHOULD populate `.ips`
+              with virtual IP address(es) where the service can be reached from within
+              the importing cluster. These addresses MAY be endpoints directly reachable
+              over a "flat" network between clusters, or MAY direct traffic through
+              topology such as an intermediate east-west gateway.
+              If exported Services of the same name and namespace in a given ClusterSet
+              have differing types, a "Conflict" status condition SHOULD be reported in
+              ServiceExport status.
+            enum:
+            - ClusterSetIP
+            - Headless
+            type: string
+        required:
+        - ports
+        - type
+        type: object
+    served: true
+    storage: false
```

Comment:
This file does not specify any webhook conversion config so who is the intended consumer?

Reply:
I wanted to figure out webhooks as a later steps when we would be actually storing v1alpha2, but we also said at the last sig meeting that we could provide only the webhook code and not the deployment manifests so I suppose that implementation would have to update that aspect if they want to have a webhook. TBD when we would be actually implementing that and figured out recomendations to integrate it to implementations.

Reply:
I can work on the webhook code - shouldn't be difficult.

Reply:
Sure that would be great! Note that I would like to submit the CoreDNS PR to v1alpha2 as soon as possible with the hope that it gets merged before the winter holidays by the CoreDNS folks. So I would recommend to not block the v1alpha2 version based on having the webhook or not.